Using the attester and verifier on two different devices

Asiderr opened the pull request:
The PR is connected to the following issue https://github.com/Fraunhofer-SIT/charra/issues/15.
These changes allow the attester to obtain the public part of the attestation key. The public key is marshaled and sent to the verifier. The verifier loads the external public key and uses it to verify the TPM quote signature.

eckelmeckel commented:
Hi @Asiderr,

that's a great improvement to CHARRA. Thanks a lot for that. I have been working on major changes to CHARRA (#17, no more malloc's; PCR verification, etc.) for the last couple of month but haven't been able to push them before today. Unfortunately, your improvements cannot be merged automatically anymore. Would you be so kind to adapt your code to the latest CHARRA changes so that it can be merged automatically? That would be great.

Again, that's a great new feature :-)

Thanks,
Michael

Commit:
attester: Send public key to the verifier

Obtain public key that is used to sign the TPM quote.
Add the key to the response and change the max size of
the PDU data.

Signed-off-by: Norbert Kamiński <norbert.kaminski@3mdeb.com>

Commit:
verifier.c: Load external public key and verify signature

Verifier unmarshalls attester public key and creates
new key handler. Then it checks the if TPM signature
is valid.

Signed-off-by: Norbert Kamiński <norbert.kaminski@3mdeb.com>

Asiderr commented:
Hi @eckelmeckel,

I've rebased the changes to the current master, but wasn't able to check if it works due to https://github.com/Fraunhofer-SIT/charra/issues/18

eckelmeckel committed:
Fixing some errors.

Signed-off-by: Michael Eckel <michael.eckel@sit.fraunhofer.de>

eckelmeckel commented:
Hi @Asiderr,

I fixed some minor errors in your PR. Unfortunately, I was not able to push it to your repo, so I put it here: https://github.com/Fraunhofer-SIT/charra/tree/docker_split

Can you please merge this branch here? Then I can rebase and merge it to master.

Thanks,
Michael

Commit:
README.md: Add documetation for remote attestation

Signed-off-by: Norbert Kamiński <norbert.kaminski@3mdeb.com>

Asiderr commented:
Hi Michael,

I rebased 3mdeb/docker_split to the current master, and I cherry-picked your fixes. Thanks for correcting my mistakes. I've tested the remote attestation, and I can confirm that everything works fine. Also, I added the instructions on how to reproduce my results to the `README.md`. 

Best regards,
Norbert

eckelmeckel reviewed (approved):
Hi Norbert (@Asiderr),

I have also successfully tested it locally. Thanks for the contribution :-)

Michael
